tests/ir/circuit/test_region_methods.py

```diff
@@ -338,3 +339,59 @@ def test_with_fold(self, r6_qudit_circuit: Circuit) -> None:
         region = r6_qudit_circuit.surround((cycle, qudit), 4)
         r6_qudit_circuit.fold(region)
         assert r6_qudit_circuit.get_unitary() == utry
+
+    def test_surround_symmetric(self) -> None:
+        circuit = Circuit(6)
+        # whole wall of even
+        circuit.append_gate(CNOTGate(), [0, 1])
+        circuit.append_gate(CNOTGate(), [2, 3])
+        circuit.append_gate(CNOTGate(), [4, 5])
+
+        # one odd gate; problematic point in test
+        circuit.append_gate(CNOTGate(), [3, 4])
+
+        # whole wall of even
+        circuit.append_gate(CNOTGate(), [0, 1])
+        circuit.append_gate(CNOTGate(), [2, 3])
+        circuit.append_gate(CNOTGate(), [4, 5])
+
+        region = circuit.surround((1, 3), 4)
+        assert region.location == CircuitLocation([2, 3, 4, 5])
+
+    def test_surround_filter_hard(self) -> None:
```

Comment:
What are some other use cases for filtering? It seems like maybe a MachineModel/topology or gateset check would be a nice extra test to have.